.travis.yml

```diff
@@ -150,6 +151,8 @@ base_linux_config: &base_linux_config
   language: python
   before_install:
     - ./build-support/bin/install_aws_cli_for_ci.sh
+    # TODO(John Sirois): Get rid of this in favor of explicitly adding pyenv versions to the PATH:
+    #   https://github.com/pantsbuild/pants/issues/7601
     - pyenv global 2.7.15 3.6.7 3.7.1
```

Comment:
> I see little ambiguity to humans and to the machine with this line.

How do you know the pyenv shimming magic is even working? IE: pyenv is partially broken already so what says the shiming magic (https://github.com/pyenv/pyenv/blob/master/README.md see "3. Add pyenv init to your shell to enable shims and autocompletion.") is installed and available for the travis user?

If speed is an issue we should simply be using images on all linux shards and thus only relying on travis to checkout our code and have docker installed. The wins there are plentiful, including being able to repro any shard on your own machine.

Reply:
> How do you know the pyenv shimming magic is even working?

Because with this line, everything works. With it removed, none of the desired interpreters are discovered. Hence https://github.com/pantsbuild/pants/pull/7411, for example.

Perhaps other parts of Pyenv don't work. I don't know, as no one has audited that. But I think that's irrelevant here - the only thing we care about is getting the pre-installed interpreters discoverable, not that things like `autocompletion` work.

We can achieve this goal through either the current single line, or the more verbose `pyenv/versions` approach, so the question is simply which approach is better for this case.

> If speed is an issue we should simply be using images on all linux shards and thus only relying on travis to checkout our code and have docker installed.

Good idea for the sake of reproducible builds. If we do go this route, in the meantime, sticking with Travis's preinstalled Pythons seems to be worth it.

Reply:
> Because with this line, everything works.

OK. Well, part of the difficulty of fixing CI is that there were bits sprinkled in that had nothing to do with getting things working. Reading these global lines it's easy to assume they are cruft. Now that I have your word this was a pivotal line, I can trust its not cruft - but hopefully you see the problem here.

I'll leave the TODO to save trees and perhaps the issue can hold dissenting views / be closed by deleting associated comments in the code.

Reply:
If you end up making other changes in this PR, then I think it'd be helpful to add a comment explaining why we have these lines like this:

```
Travis preinstalls Python interpreters via Pyenv. Due to Travis's design, we must include this line to expose the interpreters to the PATH.
```

I should have included these when I added it. Alas.